### ISSUE TITLE
HMAC: Request hash does not match the calculated hash

### ISSUE THREAD

**Issue by flashjmp** [ernestre/terraform-provider-teampasswordmanager#15]
Hi there,

### Terraform Version
```
terraform --version
Terraform v1.2.5
on linux_amd64
+ provider registry.terraform.io/ernestre/teampasswordmanager v0.2.0

```
### Affected Resource(s)
Please list the resources as a list, for example:
- data "teampasswordmanager_password"

### Terraform Configuration Files
```hcl
provider "teampasswordmanager" {
  host        = "https://tpmhost/pwmanager/"
  public_key  = "<public key>"
  private_key = "<private key>"
}


data "teampasswordmanager_password" "apwd" {
  id = "12345"
}

output "pwd" {
  value = data.teampasswordmanager_password.apwd.password
}
  
```

### Debug Output
https://gist.github.com/flashjmp/caa873fa0937c8191bf86aad34451209

### Expected Behavior
output the password from TPM

### Actual Behavior
Error Message: "Error: failed to fetch password by id 59093': HMAC: Request hash does not match the calculated hash"

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

**Comment by ernestre** [comment]
Hi @flashjmp, 

Could you make sure that your `public_key` and `private_key` are correct and you're not mixing them up? You can get them from `http://<tpm_host>/index.php/user_info/api_keys`. I've just tested the example that you've provided against the latest version of TeamPasswordManager and I couldn't reproduce the issue with correct API keys.

The only way I've managed to get the same error is by using an incorect `private_key`.

Could you provide the TeamPasswordManager version that you're running?

**Comment by flashjmp** [comment]
Hi @ernestre ,
thanks for the quick reply!

Here's the version info (via API from Postman):

{
    "version": "10.138.240",
    "version_date": "2022-02-15",
    "api_version": "5"
}

I was able to recreate the request via Postman (obtaining the password via a get on the HTTP Endpoint) and that is working fine (though calculating the HMAC Hash is a pain ;-) )

**Comment by flashjmp** [comment]
HI @ernestre ,
I debugged the provider and it seems I have found the issue.

As explained [here](https://teampasswordmanager.com/docs/api-authentication/) in the section about the HMAC (Section 2.1): "1. The URL part that starts after "index.php/"."

in auth.go in the "generateAuthHash" function you use the full endpoint (in my case "pwmanager/index.php/api/v5/passwords/12345.json"). When omitting the part up until index.php it works...
